hogql_parser/parser.cpp

```diff
@@ -1373,7 +1373,24 @@ class HogQLParseTreeConverter : public HogQLParserBaseVisitor {
     RETURN_NEW_AST_NODE("CompareOperation", "{s:N,s:N,s:N}", "left", left, "right", null_constant, "op", op);
   }
 
-  VISIT_UNSUPPORTED(ColumnExprTrim)
+  VISIT(ColumnExprTrim) {
+    const char* name;
+    if (ctx->LEADING()) {
+      name = "trimLeft";
+    } else if (ctx->TRAILING()) {
+      name = "trimRight";
+    } else if (ctx->BOTH()) {
+      name = "trim";
+    } else {
+      throw ParsingException("Unsupported value of rule ColumnExprTrim");
+    }
+    auto string_literal_terminal = ctx->STRING_LITERAL();
+    if (!string_literal_terminal) throw PyInternalException();
```

Comment:
This should be a `ParsingException`. `PyInternalException` is reserved for cases where the Python runtime has set an error, which happens inside Python/C API calls – the first such call in this visitor is `build_ast_node` down below.

But also the `STRING_LITERAL` is non-optional in the syntax, so the visitor would be practically just as safe without `if (!string_literal_terminal)`.